### ISSUE TITLE
Stuck on Awaiting Scripts

### ISSUE THREAD

**Issue by zRennox** [fivem-ex/esx_kashacter#15]
Help ;D Did everything from the tutorial.

**Comment by adamshcao** [comment]
> Help ;D Did everything from the tutorial.

Same, no errors, can't seem to find the problem.

**Comment by ScaredDonut** [comment]
Same issue, after replacing the code as stated on the readme the server hangs on awaiting scripts.

**Comment by Senlar** [comment]
Same

**Comment by Fantastic-Dave** [comment]
ive actually got an error

SCRIPT ERROR: @esx_kashacter/server/main.lua:102: attempt to index a nil value (field '?')
> handler (@esx_kashacter/server/main.lua:38)

ill post back once ive fixed it

**Comment by gunsela92** [comment]
same here some people stuck some people doesnt

**Comment by RikoDEV** [comment]
This looks like an error when getting the player's position from the database.

https://github.com/fivem-ex/esx_kashacter/blob/c1cb59fac8a0e8b4a4d711e50cb0ed37c1660ec0/esx_kashacters/server/main.lua#L109

**Comment by gunsela92** [comment]
There is no errors on console or client. Just blurred screen. Do you have any fixes ?

**Comment by RikoDEV** [comment]
Currently no.

**Comment by Fantastic-Dave** [comment]
this fixed it for me 
local SpawnPos = MySQLAsyncExecute("SELECT `position` FROM `users` WHERE `identifier` = '"..GetRockstarID(source).."'")

**Comment by RikoDEV** [comment]
Well, it shouldn't. Coordinates should be taken from the database based on the selected form. E.g. Char1, Chart2... In your case it works because you left the default code es_extended.

Our identifiers have **license:**/**CharX:** before.

**Comment by neoreturns** [comment]
Same issue, have followed everything, double checked.

Character select panels only appear after restarting resource
After selecting, the screen remains in the sky

Must be some issue directly after with spawning, looking into it also, thanks!

**Comment by Raptoxx** [comment]
Hi guys, I fixed it by writing this:
RegisterServerEvent("kashactersS:CharacterChosen")
AddEventHandler('kashactersS:CharacterChosen', function(charid, ischar)
    local src = source
    local spawn = {}
    SetLastCharacter(src, tonumber(charid))
    SetCharToIdentifier(GetPlayerIdentifiers(src)[1], tonumber(charid))
    if ischar == true then
        spawn = GetSpawnPos(src)
    else
		TriggerClientEvent('skinchanger:loadDefaultModel', src, true, cb)
        --spawn = { x = 195.55, y = -933.36, z = 29.90 } -- DEFAULT SPAWN POSITION
        spawn = { x = -530.29, y = -219.24, z = 37.65 }
    end
    TriggerClientEvent("kashactersC:SpawnCharacter", src, spawn)
end)


function GetSpawnPos(source)
    local SpawnPos = MySQLAsyncExecute("SELECT `position` FROM `users` WHERE `identifier` = '"..GetPlayerIdentifiers(source)[1].."'")
    if SpawnPos[1] ~= nil and SpawnPos[1].position ~= nil and SpawnPos[1].position ~= '' then
        return json.decode(SpawnPos[1].position)
    else
        spawn = { x = -530.29, y = -219.24, z = 37.65 }
        return spawn
    end
end

The code of the kashacters works for the characters already created, to those not created it gave "Awaiting Scripts"

**Comment by 123455-sudo** [comment]
@Raptoxx i can use this on esx_property and es_extended

**Comment by Raptoxx** [comment]
> @Raptoxx i can use this on esx_property and es_extended

Yes, but what version of es_extended are you using? For "Awaiting Scripts" you have to change some Trigger in client.lua for your es_extended version

**Comment by 123455-sudo** [comment]
@Raptoxx 
i have 1.3.0-final 
![image](https://user-images.githubusercontent.com/70820043/125600883-c94f6565-85ea-445d-a639-9ed4552126d3.png)

**Comment by Raptoxx** [comment]
> @Raptoxx
> i have 1.3.0-final
> ![image](https://user-images.githubusercontent.com/70820043/125600883-c94f6565-85ea-445d-a639-9ed4552126d3.png)

I don't use 1.3.0-final, you have to do it yourself... I don't know

**Comment by LightningRoleplay** [comment]
What file should this be in? The client or server? I'm guessing the server.lua, but at wich line did you put it?

**Comment by Raptoxx** [comment]
> What file should this be in? The client or server? I'm guessing the server.lua, but at wich line did you put it?

Yes it is server.lua line: 27 and 107

**Comment by Thecoderduwer** [comment]
same but no error

**Comment by Thecoderduwer** [comment]
![image](https://github.com/fivem-ex/esx_kashacter/assets/114108943/2ed694cd-017a-4ffe-ad27-76c96b1fb1a0)
